vignettes/datatable-benchmarking.Rmd

```diff
@@ -17,7 +17,102 @@ h2 {
 }
 </style>
 
-This vignette documents the best ways to accurately measure data.table's performance, as well as common traps to avoid.
+```{r setup}
+library(knitr)
+opts_chunk$set(fig.width = 8, fig.height = 5)
+```
+
+```{r loadPackages}
```

Comment:
No point in adding new dependencies in vignette which doesn't even need to evaluate anything. If you use R code chunks here do not add any new dependencies. We just simplified Suggests a moment ago.

Reply:
Agreed, just making the code runnable for this discussion while we work out whether what I'm saying is worthwhile. Intend to remove these dependencies just before being merged.

Reply:
`microbenchmark` is one of these dependencies, iiuc, and it's used in this vignette, which seems reasonable.  Don't we need to add `microbenchmark` to DESCRIPTION:Suggests then?  How could that dependency be removed before merging?

Reply:
`microbenchmark` probably can't, (unless we want to avoid explicit code examples and talk more generally), but things like `magrittr` and `hutils` definitely can. `ggplot2` could be replaced by base graphics if needed.

Reply:
We can just paste the code used with microbenchmark and not use it inside R code chunk. We actually don't need to use R code chunks at all in this doc. It meant to describe good practice for benchmarking, and not running benchmarks. Joshua Ulrich is new maintainer of microbenchmark and he manage to strip ggplot2 from imports to suggests so microbenchmark is now really "micro", but still if we don't use it inside data.table I don't see the point of making our Suggests list longer again. I filled #2740 because there are still some packages that could be removed from there.

Reply:
Yes agreed. Sounds good. So this vignette will display the benchmark code and results, but not actually run the benchmark.  That makes sense, as otherwise, a potentially long running benchmark would need to run when `R CMD check` runs.  Ok. Good plan.